libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -15,6 +15,8 @@
 
 extern AP_HAL::HAL& hal;
 
+Compass Compass::frontend;
+
```

Comment:
This way also doesn't enable much changes, the changes are made to give interface like <Class>::<function_call> instead of <Class>::get_frontend()-><function_call>. for the second implementation the second is similar in complexity. but the choice to give dev the first choice was to make it more readable.